modules/perception/camera/common/camera_ground_plane.cc

```diff
@@ -226,54 +225,53 @@ bool CameraGroundPlaneDetector::DetetGround(float pitch, float camera_height,
-      ADEBUG << "ph: " << ph[0] << ", " << ph[1];
-      success = fabs(ph[0]) < params_.max_tilt_angle &&
-                ph[1] < params_.max_camera_ground_height;
-      if (success) {
-        ground_plane_tracker_->Push(ph, inlier_ratio);
-        ground_plane_tracker_->GetGround(&ph[0], &ph[1]);
-        GetGround3FromPitchHeight(k_mat, baseline_, ph[0], ph[1], &ground3);
-        FillGroundModel(ground3);
-        ADEBUG << "l tracked: " << l_[0] << ", " << l_[1] << ", " << l_[2];
-        ADEBUG << "ph tracked: " << ph[0] << ", " << ph[1];
-      }
-    }
-
+  }
+  bool success = false;
+  float inlier_ratio = 0.0f;
+  std::vector<float> ph(2, 0);
+  if (CameraGroundPlaneDetector::DetectGroundFromSamples(vd, count_vd,
+                                                         &inlier_ratio)) {
+    ADEBUG << "l: " << l_[0] << ", " << l_[1] << ", " << l_[2];
+    ground3.assign(l_, l_ + 3);
+    GetGroundPlanePitchHeight(baseline_, k_mat, ground3, &ph[0], &ph[1]);
+    ADEBUG << "ph: " << ph[0] << ", " << ph[1];
+    success = fabs(ph[0]) < params_.max_tilt_angle &&
+              ph[1] < params_.max_camera_ground_height;
     if (success) {
-      ADEBUG << "succeed with inlier ratio: " << inlier_ratio;
-      ground_is_valid_ = true;
-      return true;
-    }
-
-    // backup using last successful frame or given pitch & height
-    if (ground_plane_tracker_->GetCurTrackLength() > 0) {
+      ground_plane_tracker_->Push(ph, inlier_ratio);
       ground_plane_tracker_->GetGround(&ph[0], &ph[1]);
       GetGround3FromPitchHeight(k_mat, baseline_, ph[0], ph[1], &ground3);
       FillGroundModel(ground3);
-    } else {
-      CHECK(fabs(pitch) < params_.max_tilt_angle);
-      CHECK(camera_height < params_.max_camera_ground_height);
-      CHECK_GT(camera_height, 0.f);
-      GetGround3FromPitchHeight(k_mat, baseline_, pitch, camera_height,
-                                &ground3);
-      FillGroundModel(ground3);
+      ADEBUG << "l tracked: " << l_[0] << ", " << l_[1] << ", " << l_[2];
+      ADEBUG << "ph tracked: " << ph[0] << ", " << ph[1];
     }
-    ground_plane_tracker_->Restart();
-    return false;
   }
+
+  if (success) {
+    ADEBUG << "succeed with inlier ratio: " << inlier_ratio;
+    ground_is_valid_ = true;
+    return true;
+  }
+
+  // Backup using last successful frame or given pitch & height
+  if (ground_plane_tracker_->GetCurTrackLength() > 0) {
+    ground_plane_tracker_->GetGround(&ph[0], &ph[1]);
+    GetGround3FromPitchHeight(k_mat, baseline_, ph[0], ph[1], &ground3);
+    FillGroundModel(ground3);
+  } else {
+    CHECK(fabs(pitch) < params_.max_tilt_angle);
+    CHECK(camera_height < params_.max_camera_ground_height);
+    CHECK_GT(camera_height, 0.f);
+    GetGround3FromPitchHeight(k_mat, baseline_, pitch, camera_height,
+                              &ground3);
+    FillGroundModel(ground3);
+  }
+  ground_plane_tracker_->Restart();
+  return false;
 }
 
```

Comment:
It's making the logic more complicated by checking "success" twice.

Reply:
Good point, and itt seems we don't need success at all.